subprojects/docs/src/docs/userguide/build_cache.adoc

```diff
@@ -100,85 +100,85 @@ BUILD SUCCESSFUL
+    link:{groovyDslPath}/org.gradle.api.plugins.quality.JDepend.html[JDepend],
+    link:{groovyDslPath}/org.gradle.api.plugins.quality.Pmd.html[Pmd]
 * JaCoCo:
-    api:org.gradle.testing.jacoco.tasks.JacocoMerge[],
-    api:org.gradle.testing.jacoco.tasks.JacocoReport[]
+    link:{groovyDslPath}/org.gradle.testing.jacoco.tasks.JacocoMerge.html[JacocoMerge],
+    link:{groovyDslPath}/org.gradle.testing.jacoco.tasks.JacocoReport.html[JacocoReport]
 * Other tasks:
-    api:org.gradle.api.plugins.antlr.AntlrTask[],
-    api:org.gradle.plugin.devel.tasks.ValidateTaskProperties[],
-    api:org.gradle.api.tasks.WriteProperties[]
+    link:{groovyDslPath}/org.gradle.api.plugins.antlr.AntlrTask.html[AntlrTask],
+    link:{javadocPath}/org/gradle/plugin/devel/tasks/ValidateTaskProperties.html[ValidateTaskProperties],
+    link:{groovyDslPath}/org.gradle.api.tasks.WriteProperties.html[WriteProperties]
 
 All other built-in tasks are currently not cacheable.
 
-Some tasks, like api:org.gradle.api.tasks.Copy[] or api:org.gradle.api.tasks.bundling.Jar[], usually do not make sense to make cacheable because Gradle is only copying files from one location to another.
+Some tasks, like link:{groovyDslPath}/org.gradle.api.tasks.Copy.html[Copy] or link:{groovyDslPath}/org.gradle.api.tasks.bundling.Jar.html[Jar], usually do not make sense to make cacheable because Gradle is only copying files from one location to another.
 It also doesn't make sense to make tasks cacheable that do not produce outputs or have no task actions.
 
 [[sec:task_output_caching_cacheable_3rd_party]]
-==== Third party plugins
+=== Third party plugins
 
 There are third party plugins that work well with the build cache.
 The most prominent examples are the https://developer.android.com/studio/releases/gradle-plugin.html[Android plugin 3.1+] and the https://blog.gradle.org/kotlin-build-cache-use[Kotlin plugin 1.2.21+].
 For other third party plugins, check their documentation to find out whether they support the build cache.
 
 [[sec:task_output_caching_inputs]]
-==== Declaring task inputs and outputs
+=== Declaring task inputs and outputs
 
 It is very important that a cacheable task has a complete picture of its inputs and outputs, so that the results from one build can be safely re-used somewhere else.
 
 Missing task inputs can cause incorrect cache hits, where different results are treated as identical because the same cache key is used by both executions.
 Missing task outputs can cause build failures if Gradle does not completely capture all outputs for a given task.
 Wrongly declared task inputs can lead to cache misses especially when containing volatile data or absolute paths.
-(See <<sec:task_inputs_outputs>> on what should be declared as inputs and outputs.)
+(See <<more_about_tasks.adoc#sec:task_inputs_outputs,the section called "Task inputs and outputs">> on what should be declared as inputs and outputs.)
```

Comment:
We refer here to `the section called "Task inputs and outputs` and further only to `up-to-date checks` (no section prefix). Did you decide on a case by case basis what to use? The previous version always used `the section called "..."`.

Reply:
Yes, I tried to make whatever I thought fit most naturally. Often "the section called ..." didn't fit with the rest of the sentence.

Reply:
👍